aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -155,6 +207,31 @@ public void deleteFiles(Iterable<String> paths) throws BulkDeletionFailureExcept
     }
   }
 
+  private void doSoftDelete(String path) throws S3Exception {
```

Comment:
the method should now be named "tagFileToDelete"?

Reply:
Or a generic (still helper) tagFile(String path, List<String> deleteTags). the caller, in this case deleteFiles, would be responsible for passing in the deletion tags.

Reply:
I have made the changes. Named it as `tagFileToDelete(String path, Set<Tag> deleteTags)`.